source/src_io/energy_dos.cpp

```diff
@@ -669,15 +642,286 @@ void energy::perform_dos(Local_Orbital_wfc &lowf, LCAO_Hamilt &uhm)
+					out <<std::setw(2)<< "m"<<"="<<"\""<<std::setw(40)<<m1<<"\""<<std::endl;
+					out <<std::setw(2)<< "z"<<"="<<"\""<<std::setw(40)<<N1+1<<"\""<<std::endl;
+					out << ">" <<std::endl;
+					out << "<"<<"data"<<">" <<std::endl;
+					if(GlobalV::GAMMA_ONLY_LOCAL)
+					{
+						for(int ib=0;ib<GlobalV::NBANDS;ib++)
+						{
+							if (GlobalV::NSPIN==1 || GlobalV::NSPIN==2)
+								out <<std::setw(13)<< weight(is, ib*GlobalV::NLOCAL+w);
+							else if (GlobalV::NSPIN==4)
+							{
+								int w0 = w-s0;
+								out <<std::setw(13)<< weight(is, ib*GlobalV::NLOCAL+s0+2*w0)+weight(is, ib*GlobalV::NLOCAL+s0+2*w0+1);
+							}
+						}
+						out <<std::endl;
+					}
+					else
+					{
+						for(int ik=0;ik<nks;ik++)
+						{
+							for(int ib=0;ib<GlobalV::NBANDS;ib++)
+							{
+								if (GlobalV::NSPIN==1)
+									out <<std::setw(13)<< weight(ik, ib*GlobalV::NLOCAL+w);
+								else if(GlobalV::NSPIN==2)
+									out <<std::setw(13)<< weight(ik+nks*is, ib*GlobalV::NLOCAL+w);
+								else if (GlobalV::NSPIN==4)
+								{
+									int w0 = w-s0;
+									out <<std::setw(13)<< weight(ik, ib*GlobalV::NLOCAL+s0+2*w0)+weight(ik, ib*GlobalV::NLOCAL+s0+2*w0+1);
+								}
+							}
+							out <<std::endl;
+						}
+					}
+					out << "<"<<"/"<<"data"<<">" <<std::endl;
+					out << "<"<<"/"<<"orbital"<<">" <<std::endl;
+				}//j
+			}//i
+
+			out << "<"<<"/"<<"pband"<<">" <<std::endl;
+			out.close();}
+		}//is
+		{  std::stringstream os;
+			 os<<GlobalV::global_out_dir<<"Orbital";
+			 std::ofstream out(os.str().c_str());
+			 out<< std::setw(5)<<"io"<< std::setw(8) <<"spec" <<std::setw(5)<<"l"<<std::setw(5)<<"m"<<std::setw(5)<<"z"<<std::setw(5)<<"sym"<<std::endl;
+
+
+			 for (int i=0; i<GlobalC::ucell.nat; i++)
+			 {
+				 int   t = GlobalC::ucell.iat2it[i];
+				 Atom* atom1 = &GlobalC::ucell.atoms[t];  
+				 for(int j=0; j<atom1->nw; ++j)
+				 {
+					 const int L1 = atom1->iw2l[j];
+					 const int N1 = atom1->iw2n[j];
+					 const int m1 = atom1->iw2m[j];
+					 out <<std::setw(5) << i << std::setw(8) 
+						<< GlobalC::ucell.atoms[t].label <<std::setw(5)
+							<<L1<<std::setw(5) <<m1<<std::setw(5)<<N1+1<<std::setw(15)<< ModuleBase::Name_Angular[L1][m1] << std::endl;
+				 }
+			 }
+			 out <<std::endl<<std::endl;
+			 out <<std::setw(5)<< "io"<<std::setw(2)<<"="<<std::setw(2)<<"Orbital index in supercell"<<std::endl;
+			 out <<std::setw(5)<< "spec"<<std::setw(2)<<"="<<std::setw(2)<<"Atomic species label"<<std::endl;
+			 out <<std::setw(5)<< "l"<<std::setw(2)<<"="<<std::setw(2)<<"Angular mumentum quantum number"<<std::endl;
+			 out <<std::setw(5)<< "m"<<std::setw(2)<<"="<<std::setw(2)<<"Magnetic quantum number"<<std::endl;
+			 out <<std::setw(5)<< "z"<<std::setw(2)<<"="<<std::setw(2)<<"Zeta index of orbital"<<std::endl;
+			 out <<std::setw(5)<< "sym"<<std::setw(2)<<"="<<std::setw(2)<<"Symmetry name of real orbital"<<std::endl;
+			 out.close();}
+		}//out_band
```

Comment:
Please format your code, the retraction looks wrong.

Reply:
I've done the test and the program with this part of codes can be compiled and executed successfully